types/src/v0/impls/header.rs

```diff
@@ -396,7 +399,7 @@ impl Header {
         mut state: ValidatedState,
         chain_config: ChainConfig,
         version: Version,
-        reward_distributor: Option<RewardDistributor>,
+        total_reward_distributed: Option<RewardDistributor>,
```

Comment:
Feels like the old name was better? If we only need the total passing in just that value would reduce coupling.